### PR TITLE
fix: WDS color `bgAccentSubtle` (dark mode)

### LINKED ISSUE #23870: bug: `bgAccentSubtleHover` invisible (dark mode)

### PR DESCRIPTION
**Pull request by ichik**
## Description

tl;dr Changed from naive `lighten` and `darken` to a proper color-agnostic algos

Fixes #23870 

## Media
New on the left, current on the right

https://github.com/appsmithorg/appsmith/assets/80973/640dfe12-2696-4b53-a06b-71ac19648e88

## Type of change

- Bug fix (non-breaking change which fixes an issue)
- Chore (housekeeping or task changes that don't impact user perception)

## How Has This Been Tested?

- Manual

### Test Plan
Initial POC refinement, no testing necessary

### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

File: app/client/packages/design-system/theming/src/utils/TokensAccessor/DarkModeTheme.ts [hunk 1]
```diff
@@ -109,11 +113,19 @@ export class DarkModeTheme implements ColorModeTheme {
   }
 
   private get bgAccentSubtleHover() {
-    return this.bgAccentSubtle.clone().lighten(0.06);
+    const color = this.bgAccentSubtle.clone();
```
Comment (KelvinOm) [inline app/client/packages/design-system/theming/src/utils/TokensAccessor/DarkModeTheme.ts:116]: Why don't we use the `lighten` function here?

**Review by KelvinOm — APPROVED** [review]
